cms/djangoapps/contentstore/outlines.py

```diff
@@ -0,0 +1,201 @@
+                f"but this object does not have Section field {field}."
+            )
+            raise CourseStructureError(msg)
+
+
+def _remove_version_info(usage_key):
+    """
+    When we ask modulestore for the published branch in the Studio process
+    after catching a publish signal, the items that have been changed will
+    return UsageKeys that have full version information in their attached
+    CourseKeys. This makes them hash and serialize differently. We want to
+    strip this information and have everything use a CourseKey with no
+    version information attached.
+
+    The fact that this versioned CourseKey appears is likely an unintended
+    side-effect, rather than an intentional part of the API contract. It
+    also likely doesn't happen when the modulestore is being processed from
+    a different process than the one doing the writing (e.g. a celery task
+    running on any environment other than devstack). But stripping this
+    version information out is necessary to make devstack and tests work
+    properly.
+    """
+    unversioned_course_key = usage_key.course_key.replace(branch=None, version_guid=None)
+    return usage_key.map_into_course(unversioned_course_key)
+
+
+def _make_section_data(section):
+    """
+    Generate a CourseSectionData from a SectionDescriptor.
+
+    This method does a lot of the work to convert modulestore fields to an input
+    that the learning_sequences app expects. It doesn't check for specific
+    classes (i.e. you could create your own Sequence-like XBlock), but it will
+    raise a CourseStructureError if anything you pass in is missing fields that
+    we expect in a SectionDescriptor or its SequenceDescriptor children.
+    """
+    _check_section_fields(section)
+
+    sequences_data = []
+    for sequence in section.get_children():
+        _check_sequence_fields(sequence)
+        sequences_data.append(
+            CourseLearningSequenceData(
+                usage_key=_remove_version_info(sequence.location),
+                title=sequence.display_name,
+                inaccessible_after_due=sequence.hide_after_due,
+                exam=ExamData(
+                    is_practice_exam=sequence.is_practice_exam,
+                    is_proctored_enabled=sequence.is_proctored_enabled,
+                    is_time_limited=sequence.is_time_limited,
+                ),
+                visibility=VisibilityData(
+                    hide_from_toc=sequence.hide_from_toc,
+                    visible_to_staff_only=sequence.visible_to_staff_only,
+                ),
+            )
+        )
+
+    section_data = CourseSectionData(
+        usage_key=_remove_version_info(section.location),
+        title=section.display_name,
+        sequences=sequences_data,
+        visibility=VisibilityData(
+            hide_from_toc=section.hide_from_toc,
+            visible_to_staff_only=section.visible_to_staff_only,
+        ),
+    )
+    return section_data
```

Comment:
Moved these helper functions out of `get_outline_from_modulestore`, just to make things a little easier to read.